middleware/RetryHandler.md

```diff
@@ -10,7 +10,11 @@ Provide a reusuable component that provides application developers with effectiv
 - A `retry-attempt` header SHOULD be added which contains a numeric value representing the retry number.  This value will be used for diagnostics and determining the effectiveness of the retry handler.
 - Retries MUST respect the `retry-after` response header if provided.
 - Where no `retry-after` header is provided by the server, an exponential backoff with random offset hueristic should be used to determine the retry delay.
-- Retries should be limited to a maximum value.
+- Retries should be limited to a maximum delay value. The default value for this is set at 180 seconds.
```

Comment:
We should also have a minimum max delay value to avoid customers specifying max delay values such as 0 or 0.5, which would essentially result in no retries. This minimum allowed value can be the current `DEFAULT_MAX_RETRY` value of 3.

cc/ @MIchaelMainer @darrelmiller I'm not sure whether we want this behavior since we already have a `ShouldRetry` delegate that customers should use to disable retries.

Reply:
@peombwa I believe your concern is similar to the case whereby a customer specifies a `MaxRetryTime` (max delay) value less than the `retry-after` value, in that, in both cases the `Delay` `Task` will be delayed with the server-specified `retry-after` or _exponential back-off_ delay values at least once. See the implementation: 

             cumulativeDelay = 0.0;
             while (cumulativeDelay < RetryOption.MaxRetryTime)
             {
                 Task Delay(response, retryCount, RetryOption.Delay, cancellationToken)
             }

The `cumulativeDelay` value will be incremented by the `retry-after` or _exponential back-off_ values in the call to `Task Delay()` and then fail in the subsequent retry attempt.

Reply:
I don't  think we should have a minimum delay value. We should always respect a customer specified `MaxRetryTime`. If the customer doesn't want to wait more than 500ms, then we should always respect that even if that means not retrying, or retrying once more at their specified `MaxRetryTime` (for exponential back off only - see my comment above for retry-after). Short circuiting retry before `MaxRetryTime` should only occur in `ShouldRetry`.

Reply:
I agree with @MIchaelMainer .  I don't think we should add the complexity of a minimum MaxRetryTime.  I also wonder if `RetriesTimeLimit` is a clearer name for this property.

Reply:
If we implement a default then this isn't an option unless we have a new overload that says add a timelimit and use the default value.  I'd rather just make the time limit only have an effect if one is explicitly provided.   And we shouldn't use the word "delay" to describe the TimeLimit, it is confusing.